### ISSUE TITLE
How to use cifar10 in NASBench201

### ISSUE THREAD

**Issue by AlbertiPot** [D-X-Y/NATS-Bench#27]
**Describe the Question**
A clear and concise description of the question.
- topology search space in NATS-Bench

Hi, I have some questions about how to use val and test acc when searching on cifar10 subset. 

First, when training an arch search algo, we use val acc as labels during the training procedure. And to evaluate the algo, the searched arch's test acc is queried as result. This works for cifar100 and imagenet16 subset since they offer val and test acc(I really don't know what valtest_acc for)

For cifar10, there are two subsets: cifar10-val and cifar10. 
The former offers val and test acc, so the above train and eval pipeline still works, 

![截屏2021-10-20 下午11 05 35](https://user-images.githubusercontent.com/33025006/138119659-4ecbfaa9-6c7a-4303-b592-955a62a4e391.png)

but the latter only offers test acc, so when using the latter subset, I can not build labels since there are no val acc. 

![截屏2021-10-20 下午11 06 00](https://user-images.githubusercontent.com/33025006/138119755-a5ac545e-4984-4147-b2ad-0f4982e9826c.png)

Above information is offered by api.get_more_info

Many thanks

**Comment by D-X-Y** [comment]
> Another strange thing is that when using api.show(index) function, the cifar10-val has only val acc and cifar10 has only test acc. So can I just combine using the cifar10-val and cifar10 subsets: using the val acc in cifar10-val as labels and discard its test acc, then query test acc in cifar10 as result
> 
> ![截屏2021-10-20 下午11 01 30](https://user-images.githubusercontent.com/33025006/138118991-dc210c60-3a90-4efd-8adf-d91fb38c2a5d.png)
> 
> Many many thanks

That is intended to do. As we did not want the users to use "test" on "cifar10-val".

**Comment by AlbertiPot** [comment]
Thank you so much for your quick responses.

There is another question. Since the final performance metric is the arch's test acc, but in the paper, there are two columns of metric, the test acc and val acc

![截屏2021-10-21 上午9 07 35](https://user-images.githubusercontent.com/33025006/138193162-ce4b1deb-f6f0-4d81-892a-f166f7c4cf14.png)

So how do we collect the val acc in the algo which are already as our labels. 
Is that in the evaluation stage, when the algo  get one arch, we query both its test acc and val acc?

many thanks

**Comment by D-X-Y** [comment]
The validation accuracy is using the full training recipe. During the search, you probably use some low-fidelity approximation such as training with 12 epochs, so it is not the same as the "label" that you are using.

The numbers are generated from this code (https://github.com/D-X-Y/AutoDL-Projects/blob/main/exps/NATS-Bench/draw-table.py#L184-L185), would you mind to having a try to see whether it prints 46.77?

**Comment by AlbertiPot** [comment]
Hi, thank you so much.
From my screenshot, I have confirmed that I use full training epochs with 200
![截屏2021-10-21 下午2 59 49](https://user-images.githubusercontent.com/33025006/138227731-209914d9-c955-4e13-a170-0cc57a21f678.png)

I will try the given code to double check it

Many thanks

**Comment by D-X-Y** [comment]
@AlbertiPot THanks for pointing out this issue, I also tried on my side and upload a notebook for that: https://github.com/D-X-Y/NATS-Bench/blob/main/notebooks/issue-27.ipynb
It indeed should be 46.73 instead of 46.77

**Comment by AlbertiPot** [comment]
Just check the latest version of the TPAMI version.
Sincerely appreciate your efforts in revisioning and responding to my tedious questions. Now I have a thorough understanding of the dataset. 
All the best wishes
